source/extensions/tracers/opentelemetry/tracer.cc

```diff
@@ -120,7 +121,25 @@ void Span::setAttribute(absl::string_view name, const OTelAttribute& attribute_v
   *span_.add_attributes() = key_value;
 }
 
-void Span::setTag(absl::string_view name, absl::string_view value) { setAttribute(name, value); }
+void Span::setTag(absl::string_view name, absl::string_view value) {
+  if (name == Tracing::Tags::get().HttpStatusCode) {
+    uint64_t status_code;
+    // For HTTP status codes in the 5xx range, as well as any other code the client failed to
+    // interpret, span status MUST be set to Error.
+    //
+    // For HTTP status codes in the 4xx range span status MUST be left unset in case of
+    // SpanKind.SERVER and MUST be set to Error in case of SpanKind.CLIENT.
+    if (absl::SimpleAtoi(value, &status_code)) {
+      if (status_code >= 500 ||
+          (status_code >= 400 &&
+           span_.kind() == ::opentelemetry::proto::trace::v1::Span::SPAN_KIND_CLIENT)) {
+        span_.mutable_status()->set_code(
+            ::opentelemetry::proto::trace::v1::Status::STATUS_CODE_ERROR);
+      }
+    }
+  }
+  setAttribute(name, value);
```

Comment:
Just to clarify, is it intentional to have both the attribute and the status code?
I see the benefit when introducing a new feature, just want to better understand whether this is desired.

Reply:
I confirmed with a lot of imeplements in OTel-collector, most of them set both.